### ISSUE TITLE
Adding Points to box plot

### ISSUE THREAD

**Issue by varunrd** [JohnCoene/echarts4r#401]
Hello all, I just have a small question. I am trying to plot multiple boxplots on a single chart and I want all the data points used for the box plot to be displayed on the chart. However, I am unable to do so. I tried using the e_scatter function after the e_boxplot function but it is not working as intended. Would be great if someone could help me out here. Sincerely thank you for your time.

**Comment by varunrd** [comment]
No

**Comment by JohnCoene** [comment]
Do you have some code for me to start from or something for to start from/things you tried?

**Comment by varunrd** [comment]
Hi @JohnCoene ,

This is code in plotly followed by the graph I get.
library(plotly)
plot_ly(
  data=iris,
  x = ~Species,
  y = ~Sepal.Length,
  color = ~Species,
  type= 'box',
  boxpoints = "all",
  pointpos = 0
)
![Plot_lyplot](https://user-images.githubusercontent.com/54736467/149268669-56cbd596-ec0a-4e8e-9828-33e168c4792d.png)

I am trying to make a similar plot in echarts and here is the code I have written

library(echarts4r)
library(tidyverse)
iris %>% group_by(Species) %>% 
    e_charts(Sepal.Length, id = "box") %>% 
    e_boxplot(Sepal.Length) %>% 
    e_color(color = ~Species)

The chart I get for this code is as below.
![image](https://user-images.githubusercontent.com/54736467/149270560-a92f8791-c9d9-4d95-a16e-e95b0a5ad37b.png)


I tried adding scatter points, but it did not work for me.
iris %>% group_by(Species) %>% 
  e_charts(Sepal.Length) %>% 
  e_boxplot(Sepal.Length, outliers = FALSE) %>% 
  e_scatter(Sepal.Length, symbol_size = 2, jitter_amount =2)

![image](https://user-images.githubusercontent.com/54736467/149270332-b17772d5-474f-4490-bc9d-f94683c992cb.png)

So, to sum it up, I wish to add the boxpoints as I did in plotly but I could not find such function in echarts4r. So would be great if you can provide me a solution for the same. Also I am assuming the jitter_amount  in e_scatter works vertically, is there any function to jitter it horizontally?

**Comment by lhabegger** [comment]
Hi there

I found a solution that partly works. I still face issues with getting the colors mapped correctly for the boxplots and scatterplots. This is the basic part (note the use of two x_axes and some tweaking to get the categorical and numerical axis to align. Feels hacky, but it works. Also important is the `colorBy` argument in the boxplot)
``` 
levels <- levels(as.factor(mtcars$cyl))
mtcars$levelIndices <- as.numeric(as.factor(mtcars$cyl))-0.5

mtcars |> group_by(cyl)|>
  e_charts(cyl) |> 
  e_boxplot(drat, x_index=0, colorBy="data", name="boxy", outlier=F) |>
  e_data(mtcars |> group_by(cyl),levelIndices) |>
  e_scatter(drat, symbol_size = 4, x_index=1, colorBy="series",jitter_factor = 1) |> 
  e_x_axis(index = 0, type="category", show=T, data=levels) |>
  e_x_axis(index = 1, type="value", show=F)
```
which gives output:
![image](https://user-images.githubusercontent.com/104935224/206414002-d2459d0e-ec26-4ead-bc53-eea7a6bead21.png)

As you can see the color mapping is off. In the background there are four series on the plot (one for the boxplot, three for each group in data for the scatter). The colors are assigned sequentially to the series. Of course one could define each scatter plot series separately and define colors manually, but this does not feel like the right way to go. If somebody has an idea on how to "restart" the mapping of colors from the echarts theme color palette, that would probably fix this..

```
levels <- levels(as.factor(mtcars$cyl))
mtcars$levelIndices <- as.numeric(as.factor(mtcars$cyl))-0.5

mtcars |> group_by(cyl)|>
  e_charts(cyl) |> 
  e_boxplot(drat, x_index=0, colorBy="data", outlier=F) |>
  e_data(mtcars %>% dplyr::filter(cyl == levels[1]),levelIndices) |>
  e_scatter(drat, symbol_size = 4, x_index=1, name="serie1", color="blue", jitter_factor = 1) |> 
  e_data(mtcars %>% dplyr::filter(cyl == levels[2]),levelIndices) |>
    e_scatter(drat, symbol_size = 4, x_index=1, name="serie2", color="green", jitter_factor = 1) |> 
  e_data(mtcars %>% dplyr::filter(cyl == levels[3]),levelIndices) |>
    e_scatter(drat, symbol_size = 4, x_index=1, name="serie3", color="yellow", jitter_factor = 1) |> 
  e_x_axis(index = 0, type="category", show=T, data=levels) |>
  e_x_axis(index = 1, type="value",max=max(levelIndices), show=T, data=levels)
```

![image](https://user-images.githubusercontent.com/104935224/206414929-035b25b7-6efa-416f-97c7-acb43c4b9be6.png)
